tests/unit/test_session.py

```diff
@@ -389,3 +389,28 @@ def test_connection():
 
     assert session.connection == session._conn._conn
     assert session.connection == fake_snowflake_connection
+
+
+def test_connection_expiry(monkeypatch):
+    session = Session(
+        ServerConnection(
+            {"": ""},
+            mock.Mock(
+                spec=SnowflakeConnection,
+                _telemetry=mock.Mock(),
+                _session_parameters=mock.Mock(),
+                is_closed=mock.Mock(return_value=False),
+                expired=False,
+            ),
+        ),
+    )
+    with monkeypatch.context() as m:
+        m.setattr(
+            snowflake.snowpark.session,
+            "_active_sessions",
+            {session},
+        )
+        assert Session.builder.getOrCreate() is session
+        session._conn._conn.expired = True
+        m.setattr(Session.builder, "create", None)
```

Comment:
Shouldn't this be `m.setattr(Session.builder, "create", lambda:None)` instead? I also prefer if we use MagicMock and [assert_called_once](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_once) instead.

Reply:
I wanted to give monkey a try, I now agree though :)